include/NovelRT/Animation/SpriteAnimatorState.h

```diff
@@ -11,12 +11,22 @@ namespace NovelRT::Animation {
   class SpriteAnimatorState {
   private:
     std::vector<SpriteAnimatorFrame> _frames;
-    bool _shouldLoop;
+    bool _shouldLoop = false; //this is to shut the warning up since we have no ctor
     std::vector<std::tuple<std::shared_ptr<SpriteAnimatorState>, std::vector<std::function<bool()>>>> _transitions;
 
   public:
-
     inline void insertNewState(std::shared_ptr<SpriteAnimatorState> stateTarget, std::vector<std::function<bool()>> transitionConditions) {
+
+      if (stateTarget == nullptr) {
+        return;
+      }
+
+      for (size_t i = 0; i < transitionConditions.size(); i++) {
```

Comment:
What is this for? It looks like its looking for existing `nullptr`, which should never exist?

Reply:
This is what I asked you about on Discord. This loops checks each condition to see if someone passed in an invalid function, then it erases it if that's the case, allowing for normal operation to continue.

Reply:
oh, I see. it is removing them from `transitionConditions` the parameter.
long day, carry on.